compiler/rustc_middle/src/ty/relate.rs

```diff
@@ -579,10 +579,15 @@ pub fn super_relate_consts<'tcx, R: TypeRelation<'tcx>>(
     debug!("{}.super_relate_consts(a = {:?}, b = {:?})", relation.tag(), a, b);
     let tcx = relation.tcx();
 
-    // FIXME(oli-obk): once const generics can have generic types, this assertion
-    // will likely get triggered. Move to `normalize_erasing_regions` at that point.
-    let a_ty = tcx.erase_regions(a.ty());
-    let b_ty = tcx.erase_regions(b.ty());
+    let a_ty;
+    let b_ty;
+    if relation.tcx().features().adt_const_params {
+        a_ty = tcx.normalize_erasing_regions(relation.param_env(), a.ty());
```

Comment:
I believe `normalize_erasing_regions` can panic if normalization fails. Is is guaranteed that this succeeds here (otherwise you might want to use `try_normalize_erasing_regions`)?

Reply:
I think the only way it can fail is if there were other errors already. I'd rather address the ICE here if that happens and know about it than silently assume normalization can fail here.